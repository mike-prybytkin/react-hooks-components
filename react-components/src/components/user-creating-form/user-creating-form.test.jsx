import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import UserCreatingForm from './user-creating-form';

const setUp = () =>
  render(<UserCreatingForm />, {
    wrapper: BrowserRouter,
  });

describe('User Creating Form component', () => {
  it('should create snapshot', () => {
    const { asFragment } = setUp();
    expect(asFragment()).toMatchSnapshot();
  });
});
