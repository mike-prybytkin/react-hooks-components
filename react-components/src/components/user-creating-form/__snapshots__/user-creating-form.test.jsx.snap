// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Creating Form component should create snapshot 1`] = `
<DocumentFragment>
  <form
    class="create-user-form"
  >
    <div
      class="text-input"
    >
      <label
        class="text-input__label"
      >
        Name:
        <input
          class="text-input__input"
          id="name-user-form"
          maxlength="12"
          minlength="2"
          placeholder="Your Name"
          required=""
          type="text"
        />
      </label>
    </div>
    <div
      class="date-input"
    >
      <label
        class="date-input__label"
      >
        Birthday:
        <input
          class="date-input__input"
          id="b-day-user-form"
          max="2022-01-01"
          min="1922-01-01"
          type="date"
          value="2000-01-01"
        />
      </label>
    </div>
    <div
      class="select"
    >
      <label
        class="select__label"
      >
        Your monthly income:
        <select
          class="select__select"
          id="salary-user-form"
        >
          <option
            disabled=""
          >
            Please choose an option
          </option>
          <option
            value="less than 100$"
          >
            less than 100$
          </option>
          <option
            value="100-199$"
          >
            100-199$
          </option>
          <option
            value="200-299$"
          >
            200-299$
          </option>
          <option
            value="300-399$"
          >
            300-399$
          </option>
          <option
            value="400-499$"
          >
            400-499$
          </option>
          <option
            value="over 500$"
          >
            over 500$
          </option>
        </select>
      </label>
    </div>
    <div
      class="switcher-wrapper"
    >
      <p
        class="switcher-wrapper__header"
      >
        Gender:
      </p>
      <div
        class="switcher-wrapper__field"
      >
        <input
          checked=""
          id="radio-male-user-form"
          name="switch-one"
          type="radio"
          value="Male"
        />
        <label
          for="radio-male-user-form"
        >
          Male
        </label>
        <input
          id="radio-female-user-form"
          name="switch-one"
          type="radio"
          value="Female"
        />
        <label
          for="radio-female-user-form"
        >
          Female
        </label>
      </div>
    </div>
    <div
      class="checkbox"
    >
      <input
        checked=""
        class="checkbox__input"
        id="mailing-user-form"
        type="checkbox"
      />
      <label
        class="checkbox__label"
        for="mailing-user-form"
      >
        Receive mailing
      </label>
    </div>
    <div
      class="file-upload"
    >
      <label
        class="file-upload__label"
      >
        <input
          accept="image/*"
          class="file-upload__input"
          id="avatar-user-form"
          name="avatar-user-form"
          required=""
          type="file"
        />
        Add your photo
      </label>
    </div>
    <input
      class="create-user-button"
      type="submit"
      value="Create user"
    />
  </form>
</DocumentFragment>
`;
